packages/qiskit/tests/qiskit/backend/test_qiskit_connectivity.py

```diff
@@ -0,0 +1,144 @@
+
+
+class NoMap(MockConfiguration):
+    def __init__(
+        self, coupling_map: Optional[list[tuple[int, int]]], num_qubits: int
+    ) -> None:
+        # not setting the map.
+        self.num_qubits = num_qubits
+
+
+class MockBackendV1(BackendV1):  # type: ignore
+    def __init__(self, configuration: MockConfiguration) -> None:
+        self.config = configuration
+
+    def configuration(self) -> MockConfiguration:
+        return self.config
+
+    def _default_options(self):  # type: ignore
+        pass
+
+    def run(self):  # type: ignore
+        pass
+
+
+class MockBackendV2(BackendV2):  # type: ignore
+    def __init__(
+        self, coupling_map: Optional[list[tuple[int, int]]], num_qubits: int
+    ) -> None:
+        self.__coupling_map: Optional[MockCouplingMap] = None
+        self.coupling_map = MockCouplingMap(coupling_map)
+        self.num_qubits = num_qubits
+
+    @property
+    def coupling_map(self) -> MockCouplingMap:
+        if self.__coupling_map is None:
+            raise ValueError("Coupling Map has not been initialized.")
+        return self.__coupling_map
+
+    @coupling_map.setter
+    def coupling_map(self, value: MockCouplingMap) -> None:
+        self.__coupling_map = value
+
+    @property
+    def num_qubits(self) -> int:
+        return self.__num_qubits
+
+    @num_qubits.setter
+    def num_qubits(self, value: int) -> None:
+        self.__num_qubits = value
+
+    def target(self):  # type: ignore
+        pass
+
+    def max_circuits(self):  # type: ignore
+        pass
+
+    def _default_options(self):  # type: ignore
+        pass
+
+    def run(self):  # type: ignore
+        pass
+
+
+class TestQiskitConnectivityGraphV1:
+    def test_adjacency(self) -> None:
+        coupling_map = [(0, 1), (1, 0)]
+        num_qubits = 2
+
+        config = MockConfiguration(coupling_map, num_qubits)
+        backend = MockBackendV1(config)
+
+        received_graph = device_connectivity_graph(backend)
+
+        edge_list = [(0, 1)]
+        lines = [f"{a} {b}" for (a, b) in edge_list]
+
+        constructed_graph = nx.parse_edgelist(lines, nodetype=int)
+
+        assert nx.is_isomorphic(received_graph, constructed_graph)
+
+    def test_no_coupling_map(self) -> None:
+        coupling_map = [(0, 1), (1, 0)]
+        num_qubits = 2
+
+        config = NoMap(coupling_map, num_qubits)
+        backend = MockBackendV1(config)
+
+        with pytest.raises(ValueError, match="does not have a coupling map."):
+            device_connectivity_graph(backend)
+
+
+class TestQiskitConnectivityGraphV2:
+    def test_adjacency(self) -> None:
+        coupling_map = [(0, 1), (1, 0)]
+        num_qubits = 2
+
+        backend = MockBackendV2(coupling_map, num_qubits)
+
+        received_graph = device_connectivity_graph(backend)
+
+        edge_list = [(0, 1)]
+        lines = [f"{a} {b}" for (a, b) in edge_list]
+
+        constructed_graph = nx.parse_edgelist(lines, nodetype=int)
```

Comment:
I think this code works fine, but simply creating a graph as `nx.Graph([(0, 1)])` is easier to read.